src/parser.rs

```diff
@@ -222,11 +230,16 @@ impl<'a> Parser<'a> {
     /// # }
     /// ```
     pub fn new(dialect: &'a dyn Dialect) -> Self {
+        Self::new_with_options(dialect, ParserOptions::default())
+    }
+
+    pub fn new_with_options(dialect: &'a dyn Dialect, options: ParserOptions) -> Self {
```

Comment:
What would you think about following the `with_recursion_limit` builder style pattern instead of adding another `new` function. Like this:

```rust
fn with_options(mut self, options: ParserOptions) -> Self {
...
}
```
?

Reply:
Great idea